### DISCUSSION TITLE
Reaction and share buttons

### DISCUSSION THREAD

**Discussion by sandsoncosta** [hbstack/.github#88] [Ideas]
I have included reaction and sharing buttons on my blog.
![image](https://github.com/user-attachments/assets/8a5a2d41-ee85-493a-a78c-19be0624bb89)
![image](https://github.com/user-attachments/assets/c5bf4104-6866-4f69-9483-1c8e2160b95c)
I did this manually and it was extremely simple.
The only part that was manual for each published post was the share buttons part, which I included in each post.
It would be nice if there was a simpler way to set a true or false flag and include the ID like it is done with Google Adsense and Analytics.
I am using the sharethis.com platform.

**Comment by sandsoncosta** [comment]
While I was writing this idea, I was thinking that in my case, my type of content is not interesting to have these reaction buttons.
So I'm going to remove it and include the share button in its place.
![image](https://github.com/user-attachments/assets/78d26f95-07c7-4dce-901d-5a3e32ffa670)

But I still need to tag the post.
![image](https://github.com/user-attachments/assets/38d47c2c-53b2-4a97-9351-f4243461739c)
![image](https://github.com/user-attachments/assets/61d9e29e-79dc-4e6b-8b73-8a49b4e7c361)
Well... I don't see a problem, but it's an idea for a new feature.

**Reply by razonyang** [reply]
There is a built-in [social share module for blog](https://hbstack.dev/modules/blog/social-share-buttons/), I have no plan to implement others with external services for now.

**Reply by sandsoncosta** [reply]
I hadn't seen this feature... I probably just passed it by. Thanks. Much nicer than what it is currently =)

**Reply by razonyang** [reply]
NP, but that module has fewer features and social media features (such as stats) than external services.

Please use the hook instead if it doesn't meet your needs.

Of course, new social media requests are welcomed.

**Reply by sandsoncosta** [reply]
No, the native module meets my needs very well. I just want to have a share button, as I saw that there is no way to share in the PWA or in the browser on a mobile phone.
My blog is technical, so it doesn't get many views to need a count.

**Comment by razonyang** [comment]
HB has a hooks system, you can add any content to pages, and you can find all blog hooks on [here](https://hbstack.dev/modules/blog/hooks/).

To use a hook, you'll need to:

1. Declare the hook via `hugopress.modules.hb-custom.hooks` on params.yaml, there are already serveral examples, such as the `head-end` you're using.
2. And then create the template for the hook. The file name is the same as the hook name.

> It would be nice if there was a simpler way to set a true or false flag and include the ID like it is done with Google Adsense and Analytics.

You can use Hugo template syntax to achieve this.

```html
// hook template
{{ if default true .Page.Params.social_shares }}
Social share buttons here.
{{ end }}
```
